src/dash-table/reconcile/date.ts

```diff
@@ -0,0 +1,98 @@
+import { IDatetimeColumn } from 'dash-table/components/Table/props';
+import { reconcileNull } from './null';
+import { IReconciliation } from '.';
+
+// pattern and convertToMs pulled from plotly.js
+// (simplified - no international calendars for now)
+// https://github.com/plotly/plotly.js/blob/master/src/lib/dates.js
+// Note we allow timezone info but ignore it - at least for now.
+const DATETIME_REGEXP = /^\s*(-?\d{4}|\d{2})(-(\d{1,2})(-(\d{1,2})([ Tt]([01]?\d|2[0-3])(:([0-5]\d)(:([0-5]\d(\.\d+)?))?(Z|z|[+\-]\d{2}:?\d{2})?)?)?)?)?\s*$/m;
```

Comment:
We accept both positive and negative dates but do not accept any date in the range [0,1000[ and [-999,-1[

Reply:
We do accept these, they just need leading zeros to 4 digits. eg:
```
input: '  0369-11-02 15', output: '0369-11-02 15'
```

Reply:
If allow_YY===false, maybe we should interpret all years as-is?

Reply:
Why am I not seeing the comments until I refresh... ok for leading zeros. Maybe that should be documented?

Reply:
> That said, we shouldn't accept `0` either :)

In the BCE/AD counting scheme there's no year 0. But when you use negative numbers for years, there *is* a year zero, and it's the same as 1 BCE. -1 is 2BCE, etc...
```js
d=new Date();
y1 = d.setFullYear(1);
y0 = d.setFullYear(0);
yn1 = d.setFullYear(-1);
console.log((y1 - y0), (y0 - yn1))
31622400000 31536000000 // a year is ~π*10^7 sec (*10^10 ms)
```

> If allow_YY===false, maybe we should interpret all years as-is?

I think the main outcome of this would be to cause confusion for users who have 2-digit years, that we'd end up interpreting as the first century AD. I think the default should be to *only* accept 4-digit years; that's the standard in a lot of places and it's unambiguous (at least until we want to support years beyond [-9999, 9999] but we have that limit in plotly.js and AFAIK it's never come up)

Reply:
> I think the main outcome of this would be to cause confusion
Agreed. Now that I know I can do leading zeros!

Reply:
> In the BCE/AD counting scheme there's no year 0. But when you use negative numbers for years, there _is_ a year zero, and it's the same as 1 BCE. -1 is 2BCE, etc...

Part of ISO8601.. did not know that! 💡